src/github_domain.py

```diff
@@ -0,0 +1,133 @@
+# Copyright 2023 The Oppia Authors. All Rights Reserved.
+#
+# Licensed under the Apache License, Version 2.0 (the 'License');
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#      http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an 'AS-IS' BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+"""Domain objects for github entities."""
+
+from __future__ import annotations
+
+import datetime
+
+from typing import Any, Dict, List, Optional, Type
+
+
+class Assignee:
+    """A class representing an assignee of a pull request."""
+
+    def __init__(
+        self,
+        username: str,
+        assigned_on_timestamp: datetime.datetime
+    ) -> None:
+        """
+        Args:
+            username: str. GitHub username of the assignee.
+            assigned_on_timestamp: datetime. The time when the reviewer was
+                assigned.
+        """
+        self.username = username
+        self.assigned_on_timestamp = assigned_on_timestamp
+
+    def get_waiting_time(self) -> str:
+        """Returns the waiting time on review.
+        The waiting time is the difference between the current time and the time when the
+        reviewer was assigned to the PR.
+
+        Returns:
+            str. The waiting time in days and hours.
+        """
+
+        delta = datetime.datetime.now(datetime.timezone.utc) - self.assigned_on_timestamp
+        days = delta.days
+        hours, _ = divmod(delta.seconds, 3600)
+        waiting_time = []
+        if days:
+            waiting_time.append(f'{days} day{"s" if days > 1 else ""}')
+
+        if hours:
+            waiting_time.append(f'{hours} hour{"s" if hours > 1 else ""}')
+
+        return ', '.join(waiting_time)
+
+    def __repr__(self) -> str:
+        return f'@{self.username} assigned on {self.assigned_on_timestamp}'
+
+
+class PullRequest:
+    """A class representing a pull request on GitHub."""
+
+    def __init__(
+        self,
+        url: str,
+        number: int,
+        author_username: str,
+        title: str,
+        assignees: List[Assignee]
+    ) -> None:
+        """Constructs a PullRequest object.
+
+        Args:
+            url: str. The url of the PR.
+            number: int. The PR number.
+            author_username: str. GitHub username of the PR Author.
+            title: str. PR title.
+            assignees: List(Assignee). List of assignees.
+        """
+        self.url = url
+        self.number = number
```

Comment:
pr_number

(number is too generic, it can refer to anything)

Reply:
updated!